### DISCUSSION TITLE
uWebSockets/examples/EchoServerThreaded.cpp

### DISCUSSION THREAD

**Discussion by sebaster44** [uNetworking/uWebSockets#1411] [Q&A] [answered]
I wrote a server like the one in the uWebSockets/examples/EchoServerThreaded.cpp example.
I have 12 threads listening on port 1600.

I have a test that connects 2000 clients at the same time and
all connections are open in the same thread. There is something I don't understand.
I would expect to receive connections in more than one thread.

Any help to understand would be greatly appreciated.
Thanks

**Comment by e3dio** [comment, marked as answer]
On Windows it will only send requests to 1 thread, you need to listen on different ports and reverse-proxy to each port on Windows. I don't use Mac but I think it was something similar. Same port only works on Linux, see https://github.com/uNetworking/uWebSockets/issues/1194

**Reply by sebaster44** [reply]
Thanks a lot, I am on Windows, I'll give a try to linux.
